### PR TITLE
#74 HasSize matcher for checking Iterables size

### PR DESCRIPTION
**Pull request by Iprogrammerr**
As described in #74, HasSize matcher was created. It encapsulates size and allows to check its value against Iterables.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#98](https://github.com/llorllale/cactoos-matchers/issues/98) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#98](https://github.com/llorllale/cactoos-matchers/issues/98) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C63314D6Z/972d854f-b97d-49fa-a026-892bc125e792); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Iprogrammerr** [comment]
@fabriciofx Done.

File: src/test/java/org/llorllale/cactoos/matchers/HasSizeTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            () -> new IterableOfInts(1),
+            new IsNot<>(new HasSize(2))
+        ).affirm();
+    }
+
+    @Test
+    public void matchesEmptyCollection() {
+        new Assertion<>(
+            "Expected size is not 0",
+            ArrayList::new,
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/HasSizeTest.java]: @Iprogrammerr Please, use `ListOf` instead.

**Comment by llorllale** [comment]
@Iprogrammerr @fabriciofx we should use `Matches` to test matchers

**Comment by Iprogrammerr** [comment]
@llorllale Refactored.

**Comment by llorllale** [comment]
@Iprogrammerr can you please pull from `master` and retrigger the travis build?

**Comment by Iprogrammerr** [comment]
@llorllale I don't understand, how pulling is supposed to do that? It will trigger nothing. Didn't you mean merge?

**Comment by llorllale** [comment]
@Iprogrammerr I just meant to update your PR with `master` without rebasing

**Comment by Iprogrammerr** [comment]
@llorllale Done.

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@Iprogrammerr failed due to [low coverage](https://codecov.io/gh/llorllale/cactoos-matchers/pull/98/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9vcmcvbGxvcmxsYWxlL2NhY3Rvb3MvbWF0Y2hlcnMvSGFzU2l6ZS5qYXZh)

**Comment by Iprogrammerr** [comment]
@llorllale Increased.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (12 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#98](https://github.com/llorllale/cactoos-matchers/issues/98) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Pull request merged**